egs/fisher_swbd/s5/local/chain/run_blstm_6h.sh

```diff
@@ -0,0 +1,173 @@
+    --ivector-dir exp/nnet3/ivectors_${train_set} \
+    --tree-dir $treedir \
+    --splice-indexes="-2,-1,0,1,2 0 0" \
+    --lstm-delay=" [-3,3] [-3,3] [-3,3] " \
+    --xent-regularize 0.1 \
+    --include-log-softmax false \
+    --num-lstm-layers 3 \
+    --cell-dim 1024 \
+    --hidden-dim 1024 \
+    --recurrent-projection-dim 256 \
+    --non-recurrent-projection-dim 256 \
+    --label-delay 0 \
+    --self-repair-scale 0.00001 \
+   $dir/configs || exit 1;
+
+fi
+
+if [ $stage -le 13 ]; then
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $dir/egs/storage ]; then
+    utils/create_split_dir.pl \
+     /export/b0{5,6,7,8}/$USER/kaldi-data/egs/fisher_swbd-$(date +'%m_%d_%H_%M')/s5c/$dir/egs/storage $dir/egs/storage
+  fi
+
+  touch $dir/egs/.nodelete # keep egs around when that run dies.
+
+  steps/nnet3/chain/train.py --stage $train_stage \
+    --cmd "$decode_cmd" \
+    --feat.online-ivector-dir exp/nnet3/ivectors_${train_set} \
+    --feat.cmvn-opts "--norm-means=false --norm-vars=false" \
+    --chain.xent-regularize 0.1 \
+    --chain.leaky-hmm-coefficient 0.1 \
+    --chain.l2-regularize 0.00005 \
+    --chain.apply-deriv-weights false \
+    --chain.lm-opts="--num-extra-lm-states=2000" \
+    --chain.left-deriv-truncate 0 \
+    --trainer.num-chunk-per-minibatch 64 \
+    --trainer.frames-per-iter 1200000 \
+    --trainer.max-param-change 2.0 \
+    --trainer.num-epochs $num_epochs \
+    --trainer.optimization.shrink-value 0.99 \
+    --trainer.optimization.num-jobs-initial 3 \
+    --trainer.optimization.num-jobs-final 16 \
+    --trainer.optimization.initial-effective-lrate 0.001 \
+    --trainer.optimization.final-effective-lrate 0.0001 \
+    --trainer.optimization.momentum 0.0 \
+    --egs.stage $get_egs_stage \
+    --egs.opts "--frames-overlap-per-eg 0" \
+    --egs.chunk-width 150 \
+    --egs.chunk-left-context 40 \
+    --egs.chunk-right-context 40 \
+    --egs.dir "$common_egs_dir" \
+    --cleanup.remove-egs $remove_egs \
+    --feat-dir data/${train_set}_hires \
+    --tree-dir $treedir \
+    --lat-dir exp/tri5a_lats_nodup_sp \
+    --dir $dir  || exit 1;
+fi
+
+if [ $stage -le 14 ]; then
+  # Note: it might appear that this $lang directory is mismatched, and it is as
+  # far as the 'topo' is concerned, but this script doesn't read the 'topo' from
+  # the lang directory.
+  utils/mkgraph.sh --self-loop-scale 1.0 data/lang_fsh_sw1_tg $dir $dir/graph_fsh_sw1_tg
+fi
+
+decode_suff=fsh_sw1_tg
+graph_dir=$dir/graph_fsh_sw1_tg
+if [ $stage -le 15 ]; then
+  iter_opts=
+  if [ ! -z $decode_iter ]; then
+    iter_opts=" --iter $decode_iter "
+  fi
+  for decode_set in eval2000 rt03; do
+      (
+      num_jobs=`cat data/${decode_set}_hires/utt2spk|cut -d' ' -f2|sort -u|wc -l`
+      steps/nnet3/decode.sh --acwt 1.0 --post-decode-acwt 10.0 \
+          --nj $num_jobs --cmd "$decode_cmd" $iter_opts \
+          --extra-left-context 50 \
+          --extra-right-context 50 \
+          --frames-per-chunk "150" \
```

Comment:
Thanks. I use a user define style now. Changes are made accordingly. 
Ready for another review.

在 2016/4/20 5:04, Vijayaditya Peddinti 写道:
>
> In egs/fisher_swbd/s5/local/chain/run_blstm_6h.sh 
> <https://github.com/kaldi-asr/kaldi/pull/657#discussion_r60310950>:
>
> > +
> > +decode_suff=fsh_sw1_tg
> > +graph_dir=$dir/graph_fsh_sw1_tg
> > +if [ $stage -le 15 ]; then
> > +  iter_opts=
> > +  if [ ! -z $decode_iter ]; then
> > +    iter_opts=" --iter $decode_iter "
> > +  fi
> > +  for decode_set in eval2000 rt03; do
> > +      (
> > +      num_jobs=`cat data/${decode_set}_hires/utt2spk|cut -d' ' -f2|sort -u|wc -l`
> > +      steps/nnet3/decode.sh --acwt 1.0 --post-decode-acwt 10.0 \
> > +          --nj $num_jobs --cmd "$decode_cmd" $iter_opts \
> > +          --extra-left-context 50 \
> > +          --extra-right-context 50 \
> > +          --frames-per-chunk "150" \
>
> you could just derive the extra*contexts as left-context+10 and 
> right-context+10
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/kaldi-asr/kaldi/pull/657/files/f9483fdcb59dbfa711df31c41c2bea90b23c142c#r60310950>
>